extensions/file-manager/src/components/directory.tsx

```diff
@@ -47,7 +47,9 @@ export function Directory(props: { path: string; ignores: GitIgnoreHelper[]; ini
         }
       }
       if (!preferences.showHiddenFiles) {
-        const statOutput = execSync(`stat -f%f '${props.path}/${f.name}'`, {
+        const fullPath = `${props.path}/${f.name}`;
+        const escapedPath = escapeShellArg(fullPath);
+        const statOutput = execSync(`stat -f%f ${escapedPath}`, {
```

Comment:
I would prefer to use a library like [nano-spawn](https://github.com/sindresorhus/nano-spawn) for parsing special characters.

Reply:
Agreed, a library like `nano-spawn` would be better for parsing special characters. This is a quick fix for now.